### DISCUSSION TITLE
Should `poetry show` command show information about the root package of the project itself?

### DISCUSSION THREAD

**Discussion by mshonichev** [python-poetry/poetry#8772] [Q&A]
Hello, everyone!

I'm not sure if it's a bug report or feature request, so I spur a discussion. :)

It seems a bit counterintuitive from the usability perspective to me when I try to run `poetry show <name-of-the-root-package>` and got `package not found` error.

If that's a design decision, then at least the error message is confusing, cause it says exactly `Package <name> not found` even after `poetry install`. 

What's more confusing is that similar pip command succeeds: `pip show <name-of-the-root-package>` shows that a package IS installed.

I've found a related feature request [[#7731](https://github.com/python-poetry/poetry/issues/7731)], but neither it has a lot enthusiasm upon it, nor about my actual case alone.

What do you think, should I file this issue alone as feature request or as bug?

Should this issue be revised at the dawn of the package-less poetry projects (1.8.0) [[#8770](https://github.com/python-poetry/poetry/issues/8770)]?

---
Steps to reproduce:

1. create a simple package, add some dependency, install the package
```
$ poetry new whatevername
Created package whatevername in whatevername
$ cd whatevername
$ poetry add typer
Using version ^0.9.0 for typer

Updating dependencies
Resolving dependencies... (0.3s)

Package operations: 1 install, 0 updates, 0 removals

  • Installing typer (0.9.0)

Writing lock file

$ poetry install
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: whatevername (0.1.0)
```

now, `show` the package and dependency via both `pip` and `poetry`

show pip:
```
$ pip show typer
Name: typer
Version: 0.9.0
Summary: Typer, build great CLIs. Easy to code. Based on Python type hints.
Home-page: https://github.com/tiangolo/typer
Author: Sebastián Ramírez
Author-email: tiangolo@gmail.com
License: 
Location: /private/tmp/test-data/.data/lib/python3.11/site-packages
Requires: click, typing-extensions
Required-by: whatevername

$ pip show whatevername
Name: whatevername
Version: 0.1.0
Summary: 
Home-page: 
Author: Max Shonichev
Author-email: m.shonichev@github.com
License: 
Location: /private/tmp/test-data/.data/lib/python3.11/site-packages
Editable project location: /private/tmp/test-data/whatevername
Requires: idna, typer
Required-by: 
```

show poetry:
```
$ poetry show typer
 name         : typer                                                              
 version      : 0.9.0                                                              
 description  : Typer, build great CLIs. Easy to code. Based on Python type hints. 

dependencies
 - click >=7.1.1,<9.0.0
 - typing-extensions >=3.7.4.3

$ poetry show whatevername
Package whatevername not found
```

NB: Reproduces as well even after `poetry build && pip install  dist/whatevername-0.1.0-py3-none-any.whl`

**Comment by radoering** [comment]
> What do you think, should I file this issue alone as feature request or as bug?

I think there are arguments for both:

- feature request: the project itself is special and `poetry show` might be designed to show dependencies
- bug: the documentation says `poetry show` shows packages (not dependencies)

I'd probably tend to feature request because the documentation about poetry show is not that good anyway (also see last point).

> Should this issue be revised at the dawn of the package-less poetry projects (1.8.0) [https://github.com/python-poetry/poetry/issues/8770]?

That's a very good point. I assume in non-package mode the behavior of `poetry show` should not change; in package mode the feature request is still valid.

To put it in a nutshell, I think a feature request might make sense. As always, of course, it has to be important enough for someone to implement it.

> NB: Reproduces as well even after poetry build && pip install  dist/whatevername-0.1.0-py3-none-any.whl

By the way, `poetry show` does not show the installed version but the locked version of a package (and the color shows if it's installed or not). Therefore, it's clear that a `pip install` of an abitrary package has no influence on `poetry show`. Admittedly, the documentation is quite vague about that and could use some improvements.

**Reply by mshonichev** [reply]
> By the way, `poetry show` does not show the installed version but the locked version of a package (and the color shows if it's installed or not).

oops, that explains a lot. 

I'll think a documentation pitch is really what is required here, introducing new behaviour without actual community needs is a no go
